resources/migrations/000_migrations.yaml

```diff
@@ -13933,6 +13933,60 @@ databaseChangeLog:
                   name: context
             indexName: idx_query_execution_context
 
+  - changeSet:
+      id: v46.00-045
+      author: calherries
+      comment: Added 0.46.0 -- add public_uuid to action.
+      changes:
+        - addColumn:
+            tableName: action
+            columns:
+              - column:
+                  name: public_uuid
+                  type: char(36)
+                  remarks: 'Unique UUID used to in publically-accessible links to this Action.'
+                  constraints:
+                    unique: true
+
+  - changeSet:
+      id: v46.00-055
+      author: calherries
+      comment: Added 0.46.0 -- add made_public_by_id
+      changes:
+        - addColumn:
+            tableName: action
+            columns:
+              - column:
+                  name: made_public_by_id
+                  type: int
+                  remarks: 'The ID of the User who first publically shared this Action.'
+
+  - changeSet:
+      id: v46.00-056
+      author: calherries
+      comment: Added 0.46.0 -- add public_uuid and made_public_by_id to action. public_uuid is indexed
+      changes:
+        - createIndex:
+            tableName: action
+            indexName: idx_action_public_uuid
+            columns:
+              column:
+                name: public_uuid
+
+  - changeSet:
+      id: v46.00-058
+      author: calherries
+      comment: Added 0.46.0 -- add FK constraint for action.made_public_by_id with core_user.id
+      changes:
+        - addForeignKeyConstraint:
+            baseTableName: action
+            baseColumnNames: made_public_by_id
+            referencedTableName: core_user
+            referencedColumnNames: id
+            constraintName: fk_action_made_public_by_id
+            onDelete: CASCADE
```

Comment:
deleting a user also deletes the actions they made public? That seems surprising

Reply:
It has to. We don't have any notion of a fallback user.
This is also true for `report_card`: cards will be deleted when the user is deleted

```yaml
  - changeSet:
      id: 240
      author: camsaul
      comment: Added 0.36.0
      changes:
        - addForeignKeyConstraint:
            baseTableName: report_card
            baseColumnNames: creator_id
            referencedTableName: core_user
            referencedColumnNames: id
            constraintName: fk_card_ref_user_id
            onDelete: CASCADE
```

This all works under the assumption that users are not really deleted in practice, just marked as inactive:

```clojure
(api/defendpoint-schema DELETE "/:id"
  "Disable a `User`.  This does not remove the `User` from the DB, but instead disables their account."
  [id]
  (api/check-superuser)
  (api/check-500 (db/update! User id, :is_active false))
  {:success true})
```